src/mixins/name-request-mixin.ts

```diff
@@ -131,4 +122,14 @@ export default class NameRequestMixin extends Mixins(DateMixin, LegalApiMixin) {
         return 'Conversion'
     }
   }
+
+  /** Map Nr specific types to Lear recognized entity types. */
+  mapNameRequestType (nrType: CorpTypeCd): CorpTypeCd {
+    switch (nrType) {
+      case CorpTypeCd.BC_COMPANY: return CorpTypeCd.BENEFIT_COMPANY
+      case CorpTypeCd.BC_CORPORATION: return CorpTypeCd.BC_COMPANY
+      case CorpTypeCd.NR_SOLE_PROP: return CorpTypeCd.SOLE_PROP
+      default: return nrType
+    }
+  }
```

Comment:
I am not sure about this. @severinbeauvais did we not changed the approach to use legalType instead of entity_type_cd.

Reply:
Oh nice! 
Will look into this, thank you :)

Reply:
After looking into this, the `legalType` in the nr response doesn't exactly match up with what lear would expect. 

For example, a change of name Sole Prop NR comes back as `CSP` and for General Partners `CGP`. Our system recognizes these types as `SP` and `GP`.

TLDR: some mapping is still going to be required, which property is the best to map i'm not sure so unless the old property is going to become deprecated, i think i will leave this mapping?

Reply:
That's weird. I'm using a NR to create a sole prop and it's coming across as SP:

![image](https://user-images.githubusercontent.com/10002889/159966573-50809013-4949-40e6-ad16-1c909025802d.png)

CSP = CHANGE of SP ?

Reply:
Yeah CSP and CGP

Reply:
After a discussion offline, we found a solution. Will update when api changes have been implemented! 🔥